### PR TITLE
Fixes #5: Remove dependency on truffleHogRegexes

### LINKED ISSUE #5: Remove dependency: truffleHogRegexes
Integrate regex list into base project until we have clear definition of how centralized whitelist/blacklist/regexlist will work

### PR DESCRIPTION
**Pull request by tarkatronic**
To help us get this pull request reviewed and merged quickly, please be sure to include the following items:

* [X] Tests (if applicable)
* [ ] Documentation (if applicable)
* [X] Changelog entry
* [X] A full explanation here in the PR description of the work done

## PR Type
What kind of change does this PR introduce?

* [ ] Bugfix
* [X] Feature
* [ ] Code style update (formatting, local variables)
* [X] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [ ] Tests
* [ ] Other

## Backward Compatibility

Is this change backward compatible with the most recently released version? Does it introduce changes which might change the user experience in any way? Does it alter the API in any way?

* [X] Yes (backward compatible)
* [ ] No (breaking changes)

## Issue Linking

Fixes #5 

## What's new?

Removed the dependency on `truffleHogRegexes`! Everything is now stored in `tartufo/data/default_regexes.json`. This did have one nice side-effect in that these are never even loaded from disk if you disable default regexes. Plus, now we have full control of our default regexes and we can add all that we want! This makes requests like #182 far easier for us to handle.

### CODE REVIEW

**Comment by rbailey-godaddy** [comment]
Didn't you fix this obnoxious windows+tox behavior yesterday, or are these branches all from prior to that work?

**Comment by tarkatronic** [comment]
> Didn't you fix this obnoxious windows+tox behavior yesterday, or are these branches all from prior to that work?

I did, but the fix hasn't been merged yet -- #196 is still waiting on another approval. Once that's merged, I'll get my other PRs rebased to fix that up. 😄

**Review by rbailey-godaddy — APPROVED** [review]
LGTM

**Review by sjacoby-godaddy — APPROVED** [review]

**Pull request merged**
